### ISSUE TITLE
zrepl don't keep the snapshots from same time

### ISSUE THREAD

**Issue by colttt** [zrepl/zrepl#628]
Hi,

when using zrepl it looks like it pruned snapshots on inregular order, it would be nice if we can keep all snapshots from the same time, at the moment I've that:
```
/sbin/zfs list -t snapshot |grep 20220828_
nc_storage/clouddata@wolke7_20220828_080222_000                 1.33G      -  17.8T  -
nc_storage/localdata@wolke7_20220828_011722_000                   64K      -  2.25T  -
nc_storage/mariadb@wolke7_20220828_104722_000                   1.14G      -  11.2G  -
```
that can lead to inconsistency if I restore all snapshots (cause mariadb and data are not in sync)

**Comment by problame** [comment]
Let me rephrase what you're asking for:

> When a job snapshots multiple filesystems at a given point in time, I perceive the resulting set of snapshots as a unit. Pruning should act on these units, instead of the individual per-dataset snapshots.

I'll add this to the project for snapshot management. Can't promise when that'll happen, though.
In the meantime, you can use zrepl just for replication and another tool to manage snapshots the way you like.
